File: platform-api/src/main/java/com/hyunsuk/axplatform/system/service/AiService.java
package com.hyunsuk.axplatform.system.service;

import com.hyunsuk.axplatform.system.dto.AiTestRequest;
import com.hyunsuk.axplatform.system.dto.AiTestResponse;
import lombok.AllArgsConstructor;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestClient;

@Service
@AllArgsConstructor
public class AiService {
    private final RestClient aiRestClient;

    public AiTestResponse test(AiTestRequest request) {
        System.out.println("Python 전송 DTO = " + request);

        return aiRestClient.post()
                .uri("/api/v1/ai/test")
                .contentType(MediaType.APPLICATION_JSON)
                .accept(MediaType.APPLICATION_JSON)
                .body(request)
                .retrieve()
                .body(AiTestResponse.class);
    }
}

File: ai-api/app/services/ai_service.py
from app.schemas.ai import AiTestRequest, AiTestResponse


class AiService:
    """AI 작업을 처리하는 애플리케이션 서비스."""

    async def test(self, request: AiTestRequest) -> AiTestResponse:
        return AiTestResponse(
            status="SUCCESS",
            received_message=request.message,
            result=f"Python AI API received: {request.message}",
        )


def get_ai_service() -> AiService:
    """FastAPI 의존성 주입용 팩토리."""

    return AiService()